README.md

```diff
@@ -4,6 +4,31 @@
 
 This is a meta App that provides deployment packaging for Cluster API core, bootstrap and control-plane controllers.
 
+## How it works
+
+The `make` target `generate` transfer the upstream released `cluster-api-components.yaml` into a Giant Swarm specific `HelmChart`.
+Beside the transformation into a `HelmChart` there are few other changes needed to make all the `cluster-api` components fit well into our stack.
+
+To make all the changes transparent and reproducible, `kustomize` is used to apply (mostly) all changes.
+
+Following notable commands/scripts are triggered in `make generate`:
+
+1. [`hack/generate-kustomize-patches.sh`](hack/generate-kustomize-patches.sh) 
+    1. fetch the latest upstream released manifest.
+    1. as `JSON Patch 6902` doesn't support `wildcard` patches, we iterate over all `MutatingWebhookConfiguration` and `ValidatingWebhookConfiguration` to create valid `kustomize` patches for all items in the list.
+1. `make delete-generated-helm-charts` to cleanup folder `helm/cluster-api/templates`.
+1. `kustomize build config/helm -o helm/cluster-api/templates` apply all the defined `kustomize` changes (defined in `kustomization.yaml`)
+1. [`hack/move-generated-crds.sh`](hack/move-generated-crds.sh) used to move all the `CRDs` into a own directory as `helm` isn't that good when CRDs needs to get installed.
+1. [`hack/generate-crd-version-patches.sh`](hack/generate-crd-version-patches.sh) --> `¯\_(ツ)_/¯`
```

Comment:
@nprokopic (from [this slack thread](https://gigantic.slack.com/archives/C0559SH3RJ4/p1687868292279069)) ... i'm really thinking of dropping this kustomize steps as i didn't get the benefit

Reply:
updated documentation - resolved for now